src/reference_implementations/prompting_vector_llms/llm_prompting_examples/llm_prompt_ag_news.ipynb

```diff
@@ -0,0 +1,1107 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 27,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "100%|██████████| 100/100 [03:00<00:00,  1.81s/it]\n"
+     ]
+    }
+   ],
+   "source": [
+    "prompt_template = \"From World, Sports, Business, Technology, what category does this article belong to? \"\n",
+    "prompts = [f\"{model_input_text}\\n{prompt_template}\" for model_input_text in model_input_texts]\n",
+    "# For memory management, we split the prompts into batches of size 1, since the activations are heavier.\n",
+    "prompt_batches = split_prompts_into_batches(prompts, 1)\n",
+    "predicted_labels = []\n",
+    "for prompt_batch in tqdm(prompt_batches):\n",
+    "    activations = model.get_activations(prompt_batch, [last_layer_name], short_generation_config)\n",
+    "    for activations_single_prompt in activations.activations:\n",
+    "        last_layer_matrix = activations_single_prompt[last_layer_name]\n",
+    "        predicted_label = get_label_with_highest_likelihood(\n",
+    "            last_layer_matrix, label_token_ids, int_to_label_map, right_shift=True\n",
+    "        )\n",
+    "        predicted_labels.append(predicted_label)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 28,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Prediction Accuracy: 0.56\n",
+      "Confusion Matrix with ordering ['world', 'sports', 'business', 'technology']\n",
+      "[[11  2  1 14]\n",
+      " [ 8 11  0  2]\n",
+      " [ 4  0  7 12]\n",
+      " [ 1  0  0 27]]\n",
+      "========================================================\n",
+      "Label: world, F1: 0.4230769230769231, Precision: 0.4583333333333333, Recall: 0.39285714285714285\n",
+      "Label: sports, F1: 0.6470588235294118, Precision: 0.8461538461538461, Recall: 0.5238095238095238\n",
+      "Label: business, F1: 0.4516129032258065, Precision: 0.875, Recall: 0.30434782608695654\n",
+      "Label: technology, F1: 0.6506024096385543, Precision: 0.4909090909090909, Recall: 0.9642857142857143\n"
+     ]
+    }
+   ],
+   "source": [
+    "report_metrics(predicted_labels, ag_news_labels, labels_order=[\"world\", \"sports\", \"business\", \"technology\"])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This is a large jump in the accuracy of our predictions for zero-shot. Because we're directly extracting probabilities associated with our label and selecting the highest, we have no issues with matching. Moreover, we're actually able to match the performance of few-shot prompts with just zero-shot. It should be noted that part of this is likely the fact that we're using `temperature=1.0` for the few-shot prompting, but it's still a big improvement. "
```

Comment:
Suggestions for rephrasing this:

"We can see this approach yielded a large improvement in prediction accuracy over the naive zero-shot prompting approach. Because we are only extracting the probabilities that match our label space, we do not have the issue with the prediction being outside of the label space."

Reply:
I think it can be helpful to say a bit more why we might expect `temperature=1.0` to be a cause

Reply:
Done!